src/glossary.adoc

```diff
@@ -10,21 +10,19 @@ A macro can be _fixed arity_ or _variable arity_.
 
 argument::
 A single sub-expression within a macro invocation, corresponding to one of the macro’s
-parameters.  This is by default a one-to-one relation, but a parameter's cardinality constraint
+parameters.  This is by default a one-to-one relation, but a parameter's grouping mode
 can change this to a many-to-one relation.
 
-argument cardinality::
-The number of arguments to a macro invocation associated with a particular parameter.
-Since an argument produces a stream of values, this can be different from the parameter's _stream
-cardinality_.
-
 argument group::
-In a macro invocation, a sequence containing multiple arguments, used to express parameters that
-accept more than one argument.
+The concrete syntax for a _grouped parameter_.
+In a macro invocation, a list containing multiple arguments, delimited  explicitly with `[...]`
+notation. Used to express parameters that accept more than one argument.
 
-cardinality constraint::
-The part of a macro signature that specifies a parameter's acceptable _argument cardinality_ and
-_stream cardinality_.
+cardinality::
+Describes the number of values that a parameter will accept when invoked.
```

Comment:
Dunno if this is common PL parlance, but
> ...that a parameter will accept when invoked.
sounds like the parameter itself is being invoked.

Reply:
Eh, no, its lazy. Better to say "when the macro is invoked".